docs/debugging-testing/testing/unit_testing.md

```diff
@@ -124,76 +117,40 @@ With the following steps, you can write a unit test. This example creates dummy
     #endif
     ```
 
-    **MyClass.cpp**
-
-    ```
-    #include "MyClass.h"
-    #include "OtherClass.h"
-
-    namespace example {
-
-    int MyClass::myFunction() {
-        OtherClass o = OtherClass();
-        return o.otherFunction();
-    }
-
-    }
-    ```
+1. Add a new `mbed-headers-example` interface library: 
 
-    **OtherClass.h**
+   ### Stub CMake input source file
 
+   ```
+   target_include_directories(mbed-headers-example
+    INTERFACE
+        ${MBED_PATH}/example
+        ${MBED_PATH}/example/include
+        ${MBED_PATH}/example/include/example
+    )
     ```
-    #ifndef OTHERCLASS_H_
-    #define OTHERCLASS_H_
-
-    namespace example {
 
-    class OtherClass {
-    public:
-        int otherFunction();
-    };
+1. Add the newly created `mbed-headers-example` library into exising target link library of `mbed-headers`.
 
-    }
-
-    #endif
-    ```
+1. Create the following dummy classes source in `mbed-os/example/source`:
 
-    **OtherClass.cpp**
+    **MyClass.cpp**
 
     ```
-    #include "OtherClass.h"
-
+    #include "MyClass.h"
+    
     namespace example {
 
-    int OtherClass::otherFunction() {
-        return 1;
+    int MyClass::myFunction() {
```

Comment:
`{` on a new line, just check code snippets, some use different style - should be using Mbed OS style